### PR TITLE
Mostly removes the stun mechanic from flashes, and other tweaks to flashes.

### PR DESCRIPTION
**Pull request by scheveningensun**
**Major change: The e_flash animation has been significantly extended to last much longer.** If you are flashed, or caught in the flash range of an explosion, or flashed by a flashbang without eye protection, you will be blinded for nearly 7 seconds before the effects start to wear off. This is 60% longer than previously.

The way flashes interact with various existing mechanics have also been changed. It is now possible to break a flash faster by overusing it due to the chances of it greatly increasing.

EMPs may even outright break a flash while also detonating its flash in the individual's pocket. 20% likelihood if it was used once already, or more accurately when multiplied by how many times it has been used. This can be quite significant in a pinch.

Dionaea, Vaurca, and silicons are still unaffected. Unless somehow IPCs get a pass again. Might be worth testing, because Alb and co. made IPCs have that weakness on purpose.

--
The goal of this PR is to overall reduce the amount of oppressive non-violent stunlocking in the game without having to invest significant resources, force or effort into having to stun someone. Which starts with nerfing flashes a little.

 Pepperspray isn't infinite and it's okay as it stands, especially seeing as how it can backfire. Flashes can still function as powerful utility. Flashbangs and bombs also benefit a bit from this buff against unprotected targets. Better wear flash protection!

### CODE REVIEW

**Comment by Arrow768** [comment]
You attempted to merge into master.
Changed to dev

**Comment by OneOneThreeEight** [comment]
https://forums.aurorastation.org/viewtopic.php?f=18&t=10995 Forum thread posted. Changelog will be shortly.

**Comment by Pacmandevil** [comment]
this doesn't change hitting "z" to blind the whole screen with a flash, right? That'll be significantly better than just clicking the spessmen now.

**Comment by OneOneThreeEight** [comment]
Ready for review. I wouldn't call the feedback overwhelmingly positive, but it was still fairly positive.

File: code/game/objects/items/devices/flash.dm [hunk 1]
```diff
@@ -78,15 +78,15 @@
...
-			M.Weaken(10)
 			flick("e_flash", M.flash)
 				//Vaurca damage 15/01/16
 			var/mob/living/carbon/human/H = M
 			if(isvaurca(H))
 				var/obj/item/organ/eyes/E = H.get_eyes()
 				if(!E)
 					return
-				usr << span("alert", "Your eyes burn with the intense light of the flash!.")
+				usr << span("alert", "Your eyes burn with the intense light of the flash. You keel over, your antennae twitching as you lay in a daze.")
```
Comment (LordFowl) [inline code/game/objects/items/devices/flash.dm]: I don't see this as really being necessary? Verbosity is not really desirable in attack logs.

File: code/game/objects/items/devices/flash.dm [hunk 2]
```diff
@@ -170,7 +170,7 @@
 	//It will never break on the first use.
 	switch(times_used)
 		if(0 to 5)
-			if(prob(2*times_used))	//if you use it 5 times in a minute it has a 10% chance to break!
+			if(prob(10*times_used))	//More consequential rolls are made the more you overuse the device.
```
Comment (LordFowl) [inline code/game/objects/items/devices/flash.dm]: This is an incredibly large jump. Have you tested this? According to the same math, 5 times in a minute = 50% chance to break (Although that math is dubious and I suspect it is even higher.)

Comment (OneOneThreeEight) [reply code/game/objects/items/devices/flash.dm]: Basically, here's how it works.
flash first time: doesn't break. Guaranteed to blind.
second time: 10% to break on use.
third: 20%

And so on.

File: code/game/objects/items/devices/flash.dm (hunk 1 above)
Comment (OneOneThreeEight) [reply code/game/objects/items/devices/flash.dm]: This is only what Vaurca see, it's mostly additional fluff to communicate to the bugperson "oh shit, you're blinded and stunned."

Comment (LordFowl) [reply code/game/objects/items/devices/flash.dm]: And I still do not see it as necessary. Attack logs should be simple, especially for so mundane an item.

**Comment by OneOneThreeEight** [comment]
We are now reviewer compliant. Ish.

File: code/game/objects/items/devices/flash.dm [hunk 3]
```diff
@@ -78,15 +78,15 @@
...
-			M.Weaken(10)
 			flick("e_flash", M.flash)
 				//Vaurca damage 15/01/16
 			var/mob/living/carbon/human/H = M
 			if(isvaurca(H))
 				var/obj/item/organ/eyes/E = H.get_eyes()
 				if(!E)
 					return
-				usr << span("alert", "Your eyes burn with the intense light of the flash!.")
+				usr << span("alert", "Your eyes burn with the intense light of the flash!")
```
Comment (LordFowl) [inline code/game/objects/items/devices/flash.dm]: `user` not `usr`. Ensure there are no other instances of `usr` where `user` should be used.

File: code/game/objects/items/weapons/stunbaton.dm [hunk 4]
```diff
@@ -188,7 +188,7 @@
 //secborg stun baton module
 
 /obj/item/weapon/melee/baton/robot
-	hitcost = 600
+	hitcost = 400
```
Comment (LordFowl) [inline code/game/objects/items/weapons/stunbaton.dm]: This has no bearing on the PR. PRs should be atomic, and while this is a small change the master change is contentious enough that this change should be pushed separately. Not to mention it has nothing to do with flashes. Changes to cyborg tools should be held in a different PR.

File: html/changelogs/FlashTweak.yml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+author: Scheveningen
+delete-after: True
+
+changes:
+  - balance: "Blinding sources are significantly increased in duration."
+  - rscdel: "Direct flashes no longer stun those who are -not- sentient trees, robots, or bugpeople."
+  - tweak: "Changed how flash break mechanics work. Crappy budget NT tech is more likely to break from overuse."
+  - tweak: "The security cyborg now sustains for longer with consistent use of the stun baton."
```
Comment (LordFowl) [inline html/changelogs/FlashTweak.yml]: This is a spaghetti changelog entry.

> -  tweak: "The security cyborg stun baton now consumes less energy."

Still should be moved to another PR.

Comment (OneOneThreeEight) [reply html/changelogs/FlashTweak.yml]: okiedoke.

File: code/game/objects/items/devices/flash.dm (hunk 3 above)
Comment (OneOneThreeEight) [reply code/game/objects/items/devices/flash.dm]: was there before, i didn't write it.

**Comment by OneOneThreeEight** [comment]
yey

**Comment by OneOneThreeEight** [comment]
This has been sitting for awhile.

**Comment by Arrow768** [comment]
The vote was not decisive.
It will be trialed in the upcoming dev cycle
